### PR TITLE
Added missing Circle attrs

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
Follows #2267. This PR adds the following attributes to the `Circle` class:

- `r_sqr`: representing the radius squared
- `center`: a tuple containing the x and y
- `d`, `diameter`
- `area`
- `circumference`

It also adds a new internal C api function for creating tuples more efficiently: `pg_tuple_couple_from_values_double`.

## Credits

**Geometry Project:**
For code, docs and tests:
@novialriptide @Emc2356 @itzpr3d4t0r @ScriptLineStudios @avaxar @Matiiss @newpaxonian @maqa41 @blankRiot96
Also thanks to @Starbuck5 for kickstarting the idea and the occasional help!

**Functionality added in this PR**
- `circle.c`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/circle.c
Credits to @Emc2356 @itzpr3d4t0r @novialriptide 
- `geometry_test.py`
https://github.com/pygame-community/pygame-geometry/blame/main/test/test_circle.py
Credits to @itzpr3d4t0r @novialriptide
- `geometry.rst`
https://github.com/pygame-community/pygame-geometry/blame/main/docs/circle.rst
Credits to @itzpr3d4t0r

### CODE REVIEW

**Comment by MyreMylar** [comment]
Looks like this needs stubs!

**Comment by itzpr3d4t0r** [comment]
> Looks like this needs stubs!

Oh god yeah forgot the stubs xD.

**Review by MyreMylar — APPROVED** [review]
I think we should add 'radius' as an alias for 'r' just as we have 'diameter' and 'd' - for consistency.

Otherwise it all works well and makes the `draw.circle` use case simpler:

```
circle_1 = Circle((50, 50), 30)

...

pygame.draw.circle(screen, Color("red"), circle_1.center, circle_1.r)
```

Approved 👍

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Comment by ankith26** [comment]
Hmm, this PR seems to have conflicts now

File: buildconfig/stubs/pygame/geometry.pyi [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
     x: float
     y: float
     r: float
+    radius: float
+    r_sqr: float
+    d: float
+    diameter: float
+    area: float
+    circumference: float
+    center: Tuple[float, float]
```
[the remark this answers is not in the document]
Comment (itzpr3d4t0r) [reply buildconfig/stubs/pygame/geometry.pyi]: Got it.

File: buildconfig/stubs/pygame/geometry.pyi [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+    @property
+    def radius(self) -> float: ...
+    @property
+    def r_sqr(self) -> float: ...
+    @property
+    def d(self) -> float: ...
+    @property
+    def diameter(self) -> float: ...
+    @property
+    def area(self) -> float: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/geometry.pyi:30]: These properties are missing setters?

PS: properties that are aliases in code, can also be typed as aliases in the stubs. So doing something like `radius = r` once r is defined should work

**Comment by ankith26** [comment]
I'll squash and merge but keep original co-author information intact. Thanks to all who contributed!

**Pull request merged**